x-pack/plugins/security_solution/public/common/components/data_table/index.tsx

```diff
@@ -309,19 +303,28 @@ export const DataTableComponent = React.memo<DataTableProps>(
       [dispatch, id]
     );
 
+    const columnsCellActionsProps = useMemo<UseDataGridColumnsCellActionsProps>(() => {
+      const fields: UseDataGridColumnsCellActionsProps['fields'] = disableCellActions
+        ? []
+        : columnHeaders.map((column) => ({
+            name: column.id,
+            type: column.type ?? 'keyword',
+            values: data.map(
+              ({ data: columnData }) =>
+                columnData.find((rowData) => rowData.field === column.id)?.value
+            ),
+          }));
+
+      return {
+        triggerId: CELL_ACTIONS_DEFAULT_TRIGGER,
+        fields,
```

Comment:
We need to set `scopeId` like on line 322. Otherwise, the show_top_n selection defaults to 'Raw events'.

<img width="697" alt="Screenshot 2023-02-01 at 13 11 29" src="https://user-images.githubusercontent.com/1490444/216039523-accb5cdc-9961-41ff-9ece-27fb52ffed58.png">

```suggestion
        fields,
        metadata: {
           scopeId: id,
        }
```

Note:
Configurations like this should be provided inside metadata. So the top_n component doesn't have to do `if/else` on the scope id value.